src/Symfony/Bundle/FrameworkBundle/EventListener/StartupExceptionListener.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Display a welcome page if no route is configured yet.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class StartupExceptionListener implements EventSubscriberInterface
+{
+    private $router;
+    private $projectDir;
+
+    public function __construct(RouterInterface $router, $projectDir)
+    {
+        $this->router = $router;
+        $this->projectDir = $projectDir;
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        if (!$event->isMasterRequest() || !$event->getException() instanceof NotFoundHttpException || $this->router->getRouteCollection()->count() > 0) {
```

Comment:
For this instant the collection should be initialized, so no performance issue IMHO. However what about `current($this->router->getRouteCollection())` instead?